.github/workflows/android.yml

```diff
@@ -1,58 +1,206 @@
+        run: |
+          wget -O \
+            input-sdk.tar.gz \
+            https://github.com/lutraconsulting/input-sdk/releases/download/${{ env.INPUT_SDK_VERSION }}/input-sdk-qt-${{ env.QT_VERSION }}-${{ env.INPUT_SDK_VERSION }}.tar.gz
+          tar -xvzf input-sdk.tar.gz -d ${{ github.workspace }}/input-sdk
+          ls ${{ github.workspace }}/input-sdk
 
       - name: Extract Mergin API_KEY
         env:
           MERGINSECRETS_DECRYPT_KEY: ${{ secrets.MERGINSECRETS_DECRYPT_KEY }}
         run: |
-          openssl aes-256-cbc -k $MERGINSECRETS_DECRYPT_KEY -in core/merginsecrets.cpp.enc -out core/merginsecrets.cpp -d -md md5
-
+          cd core/
+          /usr/local/opt/openssl@1.1/bin/openssl \
+              aes-256-cbc -d \
+              -in merginsecrets.cpp.enc \
+              -out merginsecrets.cpp \
+              -k "$MERGINSECRETS_DECRYPT_KEY" \
+              -md md5
 
       - name: Extract GPS keystore
         env:
           INPUTKEYSTORE_DECRYPT_KEY: ${{ secrets.INPUTKEYSTORE_DECRYPT_KEY }}
 
         run: |
-          openssl aes-256-cbc -k $INPUTKEYSTORE_DECRYPT_KEY -in Input_keystore.keystore.enc -out Input_keystore.keystore -d -md md5
+          /usr/local/opt/openssl@1.1/bin/openssl \
+           aes-256-cbc -d \
+           -in Input_keystore.keystore.enc \
+           -out Input_keystore.keystore \
+           -k $INPUTKEYSTORE_DECRYPT_KEY \
+           -md md5
+
+       # Android SDK & NDK
+      - uses: nttld/setup-ndk@v1
+        id: setup-ndk
+        with:
+          ndk-version: r21e
+          add-to-path: true
 
-      - name: Build and Upload
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+         java-version: 1.8
+
+      - name: Setup Android SDK
+        uses: malinskiy/action-android/install-sdk@release/0.1.2
+
+      - name: Install Android Tools
+        run: |
+          sdkmanager --licenses && sdkmanager --verbose "platforms;${SDK_PLATFORM}" "build-tools;${SDK_BUILD_TOOLS}" tools platform-tools
+
+      # Build Input App
+      - name: Export app/config.pri
+        run: |
+          touch ./app/config.pri
+          echo -e "android {"  >> ./app/config.pri
+          echo -e "  INPUT_SDK_PATH=${{ github.workspace }}/input-sdk"  >> ./app/config.pri
+          echo -e "  QGIS_QUICK_DATA_PATH = INPUT"  >> ./app/config.pri
+          echo -e "  QMAKE_CXX = ccache \$\$QMAKE_CXX "  >> ./app/config.pri
+          echo -e "}"  >> ./app/config.pri
+          cat ./app/config.conf
+
+      - name: Build armeabi-v7a APK
         env:
-          ARCH: ${{ matrix.arch }}
-          DROPBOX_TOKEN: ${{ secrets.INPUTAPP_BOT_DROPBOX_TOKEN }}
-          GITHUB_TOKEN: ${{ secrets.INPUTAPP_BOT_GITHUB_TOKEN }}
-          INPUTKEYSTORE_STOREPASS: ${{ secrets.INPUTKEYSTORE_STOREPASS }}
+          ARCH: armeabi-v7a
+          VERSION: ${{ env. GIT_TAG }}
+          ANDROID_NDK_ROOT: ${{ steps.setup-ndk.outputs.ndk-path }}
```

Comment:
so we will use GITHUB_RUN_NUMBER as build number? is this setup?

Reply:
CI changes will come in following PR.